update changesets (#3435)
apollographql/federation: docs/source/schema-design/federated-schemas/reference/versions.mdx

- If you maintain a [subgraph-compatible library](/graphos/reference/federation/compatible-subgraphs), consult this article to stay current with recently added directives. All of these directive definitions are also listed in the [subgraph specification](/graphos/reference/federation/subgraph-spec/#subgraph-schema-additions).

## v2.14

<hr/>

| First release | Available in GraphOS? | Minimum router version |
| ------------- | --------------------- | ---------------------- |

> | **May 2026** | **Yes** | **`2.11.0`** |

Reviewer: ```suggestion
| **May 2026** | **Yes** | **`2.14.0`** |
```

#### Composition and validation changes

#### `@interfaceObject` alongside Federation 1 subgraphs

Previously, composition failed whenever a Federation 2 subgraph used `@interfaceObject` and any Federation 1 subgraph was in the graph. The rule is now scoped to a single type name: composition reports an error only if a Federation 2 subgraph uses `@interfaceObject` on a type `T` and a Federation 1 subgraph defines `@key` on an interface that is also named `T`. `@interfaceObject` on other types can coexist with Federation 1 subgraphs when that specific conflict does not occur.

#### `@composeDirective` and custom specifications

When custom directives from the same spec are merged into the supergraph via `@composeDirective`, subgraphs on the latest version of that spec no longer need to each declare every composed directive. A directive definition must appear in at least one of those subgraphs. For example, one subgraph can link a spec, compose both `@foo` and `@bar`, and define both directives, while another subgraph links the same spec, composes only `@foo`, and defines only `@foo`.

#### `@link` usage

> Composition validates `@link` to prevent **name conflicts** between linked specs and imported definitions. Conflicting names should be disambiguated with `import:` (and `as:` when needed). If you linked `https://specs.apollo.dev/tag` or `https://specs.apollo.dev/inaccessible` as separate specs, import `@tag` and `@inaccessible` from the [`federation` specification](/graphos/reference/federation/directives) instead; relying on those standalone links was tied to behavior this release fixes.

Reviewer 2: ```suggestion
Composition validates `@link` to prevent name conflicts between linked specs and imported definitions. Disambiguate conflicting names with `import:` (and `as:` when needed). If you linked `https://specs.apollo.dev/tag` or `https://specs.apollo.dev/inaccessible` as separate specs, import `@tag` and `@inaccessible` from the [`federation` specification](/graphos/reference/federation/directives) instead; relying on those standalone links was tied to behavior that this release fixes.
```
